sympy/ntheory/factor_.py

```diff
@@ -2839,3 +2839,231 @@ def drm(n, b):
             mul *= r
         n = mul
     return n
+
+
+def is_kth_power_free(n, k=2):
+    r"""
+    Checks if a given integer `n` is `k`-th power-free.
+
+    A positive integer `n` is said to be k-th power-free if no prime factor
+    of `n` appears with an exponent greater than or equal to `k`. By default,
+    this function checks for square-freeness (k=2), meaning no prime factor
+    appears with an exponent of 2 or greater.
+
+    Parameters
+    ==========
+
+    n : int
+        A positive integer to be checked for k-th power-freeness.
+    k : int, optional
+        The value of `k` for k-th power-freeness. Defaults to 2.
+
+    Returns
+    =======
+
+    bool
+        True if `n` is k-th power-free, False otherwise.
+
+    Examples
+    ========
+
+    >>> from sympy.ntheory import is_kth_power_free
+    >>> is_kth_power_free(10)  # 10 = 2^1 * 5^1 (k=2)
+    True
+    >>> is_kth_power_free(12)  # 12 = 2^2 * 3^1 (k=2)
+    False
+    >>> is_kth_power_free(18, k=3)  # 18 = 2^1 * 3^2
+    True
+    >>> is_kth_power_free(16, k=3)  # 16 = 2^4
+    False
+
+    Notes
```

Comment:
Other functions indicate errors as follows. It might be better to standardize them.

https://github.com/sympy/sympy/blob/a7719e719c0b43ec1dbb964b01b57c4f3783be8d/sympy/ntheory/residue_ntheory.py#L40-L47